src/libmongoc/tests/test-libmongoc.c

```diff
@@ -2094,17 +2094,19 @@ _parse_server_version (const bson_t *buildinfo)
    ASSERT (bson_iter_recurse (&iter, &array_iter));
 
    /* Server returns a 4-part version like [3, 2, 0, 0], or like [3, 2, 0, -49]
-    * for an RC. Bail if number of parts is ever not 4. */
+    * for an RC. Ignore the 4th part since RCs are equivalent to non-RCs for
+    * testing purposes. */
    i = 0;
    while (bson_iter_next (&array_iter)) {
+      if (i == N_SERVER_VERSION_PARTS) {
```

Comment:
I think if the while loop above was a for loop this could be added to (or in place of, assuming the server always returns a 4-part version) the current conditional statement. This appears to be a more verbose for loop already so it shouldn't cause any changes to the logic really but should be a bit easier to reason about.

Here and below on line 2152.

Reply:
You're totally right; still have vacation brain 🤦 🌴  .

Switched both to `for` loops. The conditional statements do have two parts, one for `N_SERVER_VERSION_PARTS` and one for iterating over the BSON/string.